.github/workflows/WE2E_tests.yaml

```diff
@@ -0,0 +1,123 @@
+    outputs: 
+      run-id: ${{ steps.get_run_id.outputs.run-id}}
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          clean: false
+      - id: get_run_id
+        run: |
+          FILE="${{ github.workspace }}/run_id.txt"
+          if [[ -f "$FILE"]]; then 
+            echo "::set-output name=run-id::$(cat ${{ github.workspace}}/run_id.txt)"
+          else 
+            echo "SRWA Build failed"
+            exit 1 
+          fi
+
+  set-matrix:
+    name: Set matrix of tests from machine_suites/ci.txt 
+    needs: get-run-id
+    runs-on: self-hosted
+    if: ${{ github.event.label.name == 'ci-aws-intel-WE' }}
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+ 
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          clean: false
+      - id: set-matrix
+        run: | 
+          echo "::set-output name=matrix::$(cat ufs-srweather-app-${{ needs.get-run-id.outputs.run-id }}/regional_workflow/tests/WE2E/machine_suites/ci.txt | jq -R -s -c 'split("\n")[:-1]')"
+          echo 
+
+  create-experiment-directory:
+    needs: [ set-matrix ]
+    runs-on: self-hosted
+    if: ${{ github.event.label.name == 'ci-aws-intel-WE' }}
+    # We use THIS workflow's run id to set a new experiment directory, not the SRWA build workflow run_id 
+    steps:
+      - name: Create experiment directory 
+        run: |
+          mkdir -p /scratch1/ci-WE2E-test/${{ github.run_id }}
+
+  submit-all-tests:
+    name: Submit experiments to ./run_WE2E_tests.sh
+    needs: [ set-matrix, create-experiment-directory, get-run-id ]
+    runs-on: self-hosted
+    if: ${{ github.event.label.name == 'ci-aws-intel-WE' }}
+
+    steps:
+      - name: Submit all experiments (WE2E tests) 
+        run: | 
+          cd ufs-srweather-app-${{ needs.get-run-id.outputs.run-id }}/regional_workflow/tests/WE2E/
+          module use -a /scratch1/miniconda3/contrib_miniconda3/modulefiles
+          module load miniconda3/4.11.0
+          conda activate regional_workflow
+          ./run_WE2E_tests.sh \
+            tests_file=machine_suites/ci.txt \
+            machine=aws \
+            account=account \
+            expt_basedir=/scratch1/ci-WE2E-test/${{ github.run_id }} \
+            exec_subdir=bin_intel/bin \
+            cron_relaunch_intvl_mnts=01 \
+            build_mod_fn=build_aws_intel
+
+  check-overall-we2e-status:
+    name: Check all of the experiments were submitted
+    needs: [ set-matrix, create-experiment-directory, submit-all-tests, get-run-id ]
+    runs-on: self-hosted
+    if: ${{ github.event.label.name == 'ci-aws-intel-WE' }}
+    steps:
+      - name: Check the status of all experiments
+        run: |
+          echo "Check Workflow Generation/Status"
+          cd ufs-srweather-app-${{ needs.get-run-id.outputs.run-id }}/regional_workflow/tests/WE2E/
+          ./get_expts_status.sh expts_basedir=/scratch1/ci-WE2E-test/${{ github.run_id }}
```

Comment:
How often does this run? Seems we may want to check back in over time to see if things have changed?

Reply:
`./get_expts_status.sh` is only run once to verify that the matrix of WE2E tests was submitted. In the following job, when `./get_single_expt_status.sh` is called, that script checks the output log for each of the individual WE2E Tests. I would think it is redundant to call `./get_expts_status.sh` again on all of the tests, if we're checking each one individually.  

What advantage do see to calling `./get_expts_status.sh` again, if we are checking each WE2E Test individually?